components/server/ee/src/gitlab/gitlab-app-support.ts

```diff
@@ -61,4 +76,36 @@ export class GitLabAppSupport {
         result.unshift(...ownersRepos);
         return result;
     }
+
+    protected async getAccountAvatarUrl(project: ProjectSchema, providerHost: string): Promise<string> {
+        let owner = project.owner;
+        if (!owner && project.namespace && !project.namespace.parent_id) {
+            // Fall back to "root namespace" / "top-level group"
+            owner = project.namespace;
+        }
+        if (!owner) {
+            // Could not determine account avatar
+            return "";
+        }
+        if (owner.avatar_url) {
+            const url = owner.avatar_url;
+            // Sometimes GitLab avatar URLs are relative -- ensure we always use the correct host
+            return url[0] === "/" ? `https://${providerHost}${url}` : url;
+        }
+        // If there is no avatar, generate the same default avatar that GitLab uses. Based on:
+        // - https://gitlab.com/gitlab-org/gitlab/-/blob/b2a22b6e85200ce55ab09b5c765043441b086c96/app/helpers/avatars_helper.rb#L151-161
+        // - https://gitlab.com/gitlab-org/gitlab/-/blob/861f52858a1db07bdb122fe947dec9b0a09ce807/app/assets/stylesheets/startup/startup-general.scss#L1611-1631
+        // - https://gitlab.com/gitlab-org/gitlab/-/blob/861f52858a1db07bdb122fe947dec9b0a09ce807/app/assets/stylesheets/startup/startup-general.scss#L420-422
+        const backgroundColor = ["#fcf1ef", "#f4f0ff", "#f1f1ff", "#e9f3fc", "#ecf4ee", "#fdf1dd", "#f0f0f0"][
+            owner.id % 7
```

Comment:
```suggestion
        const backgroundColors = ["#fcf1ef", "#f4f0ff", "#f1f1ff", "#e9f3fc", "#ecf4ee", "#fdf1dd", "#f0f0f0"];
        const backgroundColor = backgroundColors[owner.id % backgroundColors.length]
```
To avoid the future fix-up when a new color is added to the mix.